battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java

```diff
@@ -0,0 +1,46 @@
+package mods.battlegear2.items.arrows;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.DamageSource;
+import net.minecraft.world.World;
+/**
+ * 
+ * @author Seitenca
+ *
+ */
+public class EntityIceArrow extends AbstractMBArrow{
+
+	public EntityIceArrow(World par1World) {
+        super(par1World);
+        isImmuneToFire = false;
+    }
+    
+    public EntityIceArrow(World par1World, EntityLivingBase par2EntityLivingBase, float par3) {
+        super(par1World, par2EntityLivingBase, par3);
+        isImmuneToFire = false;
+    }
+
+    public EntityIceArrow(World par1World, EntityLivingBase par2EntityLivingBase, EntityLivingBase par3EntityLivingBase, float par4, float par5) {
+        super(par1World, par2EntityLivingBase, par3EntityLivingBase, par4, par5);
+        isImmuneToFire = false;
+    }
+
+    @Override
+    public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
```

Comment:
You can insert a custom damage source here as discussed in the ticket

Similar to the piercing arrow, except use "Frost" as the damagesource name. 
https://github.com/GTNewHorizons/Battlegear2/blob/7e8dcdb2811f1c55f892925756f4924e2e65a529/battlegear%20mod%20src/minecraft/mods/battlegear2/items/arrows/EntityPiercingArrow.java#L51

Reply:
@Override
	public boolean onHitEntity(Entity entityHit, DamageSource Frost, float ammount) {
		if(entityHit.isImmuneToFire() == true) {
			entityHit.attackEntityFrom(getFrostDamage(), ammount);
			if(!worldObj.isRemote && entityHit instanceof IShearable){
				if(((IShearable)entityHit).isShearable(null, worldObj, (int) entityHit.posX, (int) entityHit.posY, (int) entityHit.posZ)){
					ArrayList<ItemStack> drops = ((IShearable)entityHit).onSheared(null, worldObj, (int) entityHit.posX, (int) entityHit.posY, (int) entityHit.posZ, 1);
					Random rand = new Random();
	                for(ItemStack stack : drops){
	                    EntityItem ent = entityHit.entityDropItem(stack, 1.0F);
	                    ent.motionY += rand.nextFloat() * 0.05F;
	                    ent.motionX += (rand.nextFloat() - rand.nextFloat()) * 0.1F;
	                    ent.motionZ += (rand.nextFloat() - rand.nextFloat()) * 0.1F;
	                }
				}
			}
		
		}
		return true;
	}
something like this?

Reply:
More like this

```java 
	public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
		entityHit.attackEntityFrom(new EntityDamageSourceIndirect("frost", null, shootingEntity).setProjectile().setDamageBypassesArmor(), ammount);
		return true;
	}
```

Reply:
It probably should apply some slow effect, and maybe ice particles on impact but we can add those later.

Reply:
entity that != isImmuneToFire, are inmune to these arrows.
We leave it that way or I change it

Reply:
@Override
	public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
		boolean flag = false;
		if(entityHit.isImmuneToFire() == true) {
			entityHit.attackEntityFrom(new EntityDamageSourceIndirect("frost", null, shootingEntity).setProjectile().setDamageBypassesArmor(), ammount);
		}
		if(entityHit instanceof EntityLivingBase){
			((EntityLivingBase) entityHit).addPotionEffect(new PotionEffect(Potion.moveSlowdown.id, 200));
	        ((EntityLivingBase) entityHit).setArrowCountInEntity(((EntityLivingBase) entityHit).getArrowCountInEntity()+1);
			flag = true;
		}
		setDead();
		return flag;
	}

Reply:
Looks good to me.  What's going on with the setDead();? Is that to "kill" the arrow?

Reply:
@Override
	public void onHitGround(int x, int y, int z) {
		setDead();
	}

Reply:
@Override
	public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
		boolean flag = false;
		if(entityHit.isImmuneToFire() == true) {
			entityHit.attackEntityFrom(new EntityDamageSourceIndirect("frost", null, shootingEntity).setProjectile().setDamageBypassesArmor(), 13);
		}
		if(entityHit instanceof EntityLivingBase){
			((EntityLivingBase)entityHit).attackEntityFrom(new EntityDamageSourceIndirect("default", null, shootingEntity).setProjectile(), 3);
			((EntityLivingBase)entityHit).addPotionEffect(new PotionEffect(Potion.moveSlowdown.id, 200, 2));
	        ((EntityLivingBase)entityHit).setArrowCountInEntity(((EntityLivingBase) entityHit).getArrowCountInEntity()+1);
			flag = true;
		}
		setDead();
		return flag;
	}

Reply:
Wait, does this mean someone who drinks a resistance to fire potion takes extra damage from the arrow?  
...
I'll allow it. :D

Reply:
yes hehe
13 damage if is immune else 3
what do you think?